### DISCUSSION TITLE
JPG export

### DISCUSSION THREAD

**Discussion by MaxiPoivre** [nagadomi/nunif#245] [Ideas]
Hi,
When I convert photos, the software produces PNG files.
Would it be easy to implement an export format choice ? PNG or JPG with compression factor ? 
It might be also useful for people who export the depth maps sequence, to have it in JPG for space purpose.
I easily convert the PNG to JPG with another software but it could be nice to do it directly in IW3.

**Comment by nagadomi** [comment]
For export, see #154. See https://github.com/nagadomi/nunif/discussions/149#discussioncomment-9857930 for why 16-bit is needed.
For SBS output, webp(lossless) and jpeg (fixed quality 95) can be supported.

**Reply by MaxiPoivre** [reply]
For Depth maps, I understand.
But I would appreciate the jpeg output for photo conversion if you can add it easily.

**Reply by nagadomi** [reply]
Added `Image Format` option.

**Reply by MaxiPoivre** [reply]
Thanks ! Perfect ! You are very responsive!

There's something strange with photos taken with my camera (not smarphone, but just a camera). When a photo is a Portrait, IW3 turns the photo and makes it landscape. However the thumbnail is correct (portrait) in Windows and the file properties are correct too (width<height). So everything looks normal but IW3 output is wrong (landscape). The issue is only for portrait photos. Landscapes are ok.
If I open my photo with a viewer and save it (without any modifications), then IW3 makes it correct (portrait). 
Is it something wrong with the metadata of the photo saved by my camera ?

It works correctly with photos taken with my smartphone.

**Reply by nagadomi** [reply]
Yes, mobile camera adds hardware orientation state to EXIF Orientation tag of the photo when it is taken.
Image viewers that support metadata will fix the orientation when loading images.
In iw3, you can manually rotate the images using `Video Filter > Rotate` (which works for images as well), but when processing a large number of images with various orientations, it would be better to have automatic rotation fix feature.

**Reply by nagadomi** [reply]
Added `EXIF Transpose` option (True by default).

**Reply by MaxiPoivre** [reply]
Thanks a lot ! You do an incredible job !
